starknet/src/server/stream.rs

```diff
@@ -74,6 +79,31 @@ where
 
         ResponseStream::new(data_stream).instrument(stream_span)
     }
+
+    async fn sync_status(self, rpc_url: Url) -> StatusResponse {
+        let provider = HttpProvider::new(rpc_url);
+        let latest_apibara_block = self.storage.highest_accepted_block().unwrap().unwrap();
+        let latest_starknet_block = provider.get_head().await.unwrap();
+        if latest_apibara_block.eq(&latest_starknet_block) {
+            StatusResponse {
+                finalized: None,
+                latest: None,
+                status: IngestionStatus::Synced.into(),
+            }
+        } else if latest_apibara_block.ne(&latest_starknet_block) {
+            StatusResponse {
+                finalized: None,
+                latest: None,
+                status: IngestionStatus::Syncing.into(),
+            }
+        } else {
+            StatusResponse {
+                finalized: None,
+                latest: None,
+                status: IngestionStatus::Unknown.into(),
+            }
+        }
+    }
```

Comment:
```suggestion
        let finalized = node_finalized_block.map(|id| id.to_cursor());
        if let Some(accepted_block) = node_accepted_block {
        let latest = Some(accepted_block.to_cursor());
        let status = if node_accepted_block.eq(&latest_starknet_block) {
            IngestionStatus::Synced
        } else {
            IngestionStatus::Syncing
        };
        SatusResponse {
          finalized,
          latest,
          status: status.into(),
        }
    } else {
       SatusResponse {
          finalized,
          latest: None,
          status: IngestionStauts::Syncing.into(),
        }
    }
```

Here I changed it to also return the finalized and accept block id.  Also, if the block is not synced we always return `Syncing`.